packages/rest/test/integration/rest.application.integration.ts

```diff
@@ -92,6 +127,19 @@ describe('RestApplication (integration)', () => {
       .expect('Hello');
   });
 
+  it('registers controllers defined later than static assets', async () => {
```

Comment:
It's important to run this test against `RestServer` because that's the place the bug was caused.

If you feel it's important to duplicate this test for RestApplication too, then I can live with that, but please test this scenario for bare RestServer.

Reply:
Moved to `RestServer ` integration test.